bundle/edu.gemini.qpt.client/src/main/java/edu/gemini/qpt/ui/action/PublishAction.java

```diff
@@ -82,120 +78,16 @@ protected void asyncActionPerformed(ActionEvent e) {
-//                            FormFactory.UNRELATED_GAP_COLSPEC,
-//                            new ColumnSpec("max(min;30dlu)"),
-//                    },
-//                    new RowSpec[] {
-//                            new RowSpec(RowSpec.FILL, Sizes.DLUY1, FormSpec.DEFAULT_GROW),
-//                            FormFactory.RELATED_GAP_ROWSPEC,
-//                            new RowSpec(RowSpec.FILL, Sizes.DLUY1, FormSpec.DEFAULT_GROW),
-//
-//
-//                    }));
-//            final JComboBox combo = new JComboBox(OPTIONS);
-//            contentPane.add(combo, cc.xywh(1, 1, 5, 1));
-//            JButton localBtn;
-//            localBtn = new JButton("Publish in Local Time");
-//            localBtn.addActionListener(new ActionListener() {
-//                @Override
-//                public void actionPerformed(ActionEvent actionEvent) {
-//                    utc = true;
-//                    web = combo.getSelectedIndex() == 2;
-//                    qcMarkers = combo.getSelectedIndex() == 1;
-//                    cancel = false;
-//                    latch.countDown();
-//                    dialog.dispose();
-//                }
-//            });
-//            JButton utcBtn;
-//            utcBtn = new JButton("Publish in UTC");
-//            utcBtn.addActionListener(new ActionListener() {
-//                @Override
-//                public void actionPerformed(ActionEvent actionEvent) {
-//                    utc = true;
-//                    web = combo.getSelectedIndex() == 2;
-//                    qcMarkers = combo.getSelectedIndex() == 1;
-//                    cancel = false;
-//                    latch.countDown();
-//                    dialog.dispose();
-//                }
-//            });
-//
-//            JButton cancelBtn;
-//            cancelBtn = new JButton("Cancel");
-//            cancelBtn.addActionListener(new ActionListener() {
-//                @Override
-//                public void actionPerformed(ActionEvent actionEvent) {
-//                    cancel = true;
-//                    latch.countDown();
-//                    dialog.dispose();
-//                }
-//            });
-//            contentPane.add(localBtn,cc.xywh(1,3,1,1));
-//            contentPane.add(utcBtn,cc.xywh(3,3,1,1));
-//            contentPane.add(cancelBtn,cc.xywh(5,3,1,1));
-//            dialog.setContentPane(contentPane);
-//
-//
-//            dialog.setVisible(true);
-//            try {
-//                latch.await();
-//            } catch (InterruptedException e1) {
-//                System.out.println("sldjfhskfgh");
-//            }
-//            System.out.println("utc " + utc + " web " + web + " qc " + qcMarkers + " cancel " + cancel);
-//
-//            if (cancel) {
-//                return;
-//            }
+            final scala.Option<PublishDialog.PublishOptions> res = PublishDialog.prompt(shell.getPeer(), opts);
```

Comment:
It seems this will prompt every time you publish, would it make sense to make it enter the password only once per session?